File: backend/src/main/java/com/guilledev/backend/producto/dto/ProductoResponse.java
package com.guilledev.backend.producto.dto;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class ProductoResponse {

    private Long id;

    private String nombre;

    private String descripcion;

    private BigDecimal precio;

    private Integer stock;

    private Boolean activo;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    private Long categoriaId;
    private String categoriaNombre;

    private Long marcaId;
    private String marcaNombre;

    private Long proveedorId;
    private String proveedorNombre;

    private Long tipoDeProductoId;
    private String tipoDeProductoNombre;

}

File: frontend/src/app/producto/models/producto-response.ts
export interface ProductoResponse {
    id: number;
    nombre: string;
    descripcion: string;
    precio: number;
    stock: number;
    activo: boolean;

    categoriaId: number;
    categoriaNombre: string;

    marcaId: number;
    marcaNombre: string;

    proveedorId: number;
    proveedorNombre: string;

    tipoDeProductoId: number;
    tipoDeProductoNombre: string;

    createdAt: string;
    updatedAt: string;
}